### PR TITLE
RFC: chore(generate): add swagger spec patch flow

### PR DESCRIPTION
**Pull request by Nkmol**
This PR is an RFC to define our own custom API definitions that deviate from the official swagger/OpenAPI definition of Atlassian. Continuation from my comment of https://github.com/DrFaust92/bitbucket-go-client/pull/16#issuecomment-1372742614.

**Why**
This, for example, happens with the repository slug property. This property is and **[will not](https://community.atlassian.com/t5/Bitbucket-questions/Feature-Request-Add-slug-repository-property-to-API/qaq-p/2224854#M88950)** be added to their API definition.

This is one of the many deviations we currently have. In order to keep track of those changes (and to not manually revert changes after a codegen), we can use partial swagger definitions that overwrite properties of the Atlassian swagger definition. I am using the [JSON Patch](https://community.atlassian.com/t5/Bitbucket-questions/Feature-Request-Add-slug-repository-property-to-API/qaq-p/2224854#M88950) standard to define what exactly needs to be overwritten.

This PR, as an example, defines the new slug property and _adds_ it to the final swagger definition. If we do this for all our deviations, we can 100% rely on the swagger-codegen without any code manual code interference of us.

**How to test**

Run `make patch` to see the new flow.

**Addition notes**

As we use the JSON file directly from Atlassian, there is no need for the `swagger.yaml` output generated by swagger-codegen. I was however unable to disable this as swagger-codegen does this statically (https://github.com/swagger-api/swagger-codegen/blob/751e59df060b1c3ecf54921e104f2086dfa9f820/modules/swagger-codegen/src/main/resources/go/swagger.mustache). We could delete this file and ignore it in the future, IMO.

### CODE REVIEW

File: docs/Repository.md [hunk 1]
```diff
@@ -4,24 +4,25 @@
...
-**CreatedOn** | [**time.Time**](time.Time.md) |  | [optional] [default to null]
-**UpdatedOn** | [**time.Time**](time.Time.md) |  | [optional] [default to null]
 **Size** | **int32** |  | [optional] [default to null]
-**Language** | **string** |  | [optional] [default to null]
-**HasIssues** | **bool** |  | [optional] [default to null]
-**HasWiki** | **bool** |  | [optional] [default to null]
-**ForkPolicy** | **string** |  Controls the rules for forking this repository.  * **allow_forks**: unrestricted forking * **no_public_forks**: restrict forking to private forks (forks cannot   be made public later) * **no_forks**: deny all forking  | [optional] [default to null]
-**Project** | [***Project**](project.md) |  | [optional] [default to null]
-**Mainbranch** | [***Branch**](branch.md) |  | [optional] [default to null]
+**Slug** | **string** | Slugified name, the same string that is used in Bitbucket URLs. | [optional] [default to null]
```
Comment (Nkmol) [inline docs/Repository.md:23]: Auto generated flow correctly defines the "slug" property.

File: model_repository.go [hunk 2]
```diff
@@ -14,27 +14,28 @@ import (
...
+	Links      *RepositoryLinks `json:"links,omitempty"`
+	Mainbranch *Branch          `json:"mainbranch,omitempty"`
+	Name       string           `json:"name,omitempty"`
+	Owner      *Account         `json:"owner,omitempty"`
+	Parent     *Repository      `json:"parent,omitempty"`
+	Project    *Project         `json:"project,omitempty"`
+	Scm        string           `json:"scm,omitempty"`
+	Size       int32            `json:"size,omitempty"`
+	// Slugified name, the same string that is used in Bitbucket URLs.
+	Slug      string    `json:"slug,omitempty"`
```
Comment (Nkmol) [inline model_repository.go:37]: Auto generated flow correctly defines the "slug" property.

**Comment by DrFaust92** [comment]
@Nkmol i think we may want to add api/swagger.output.json to git ignore? its needed for gen time only, or am i mistaken?

**Comment by Nkmol** [comment]
> @Nkmol i think we may want to add api/swagger.output.json to git ignore? its needed for gen time only, or am i mistaken?

Correct. I added it to the `.gitignore`

**Pull request merged**